src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -983,7 +983,16 @@ class SelectClauseSegment(BaseSegment):
     type = "select_clause"
     match_grammar = StartsWith(
         Sequence("SELECT", Ref("WildcardExpressionSegment", optional=True)),
-        terminator=OneOf("FROM", "LIMIT", Ref("SetOperatorSegment")),
+        terminator=OneOf(
+            "FROM",
+            "LIMIT",
+            # :HACK: This Sequence("UNION", ... stuff should not be needed,
```

Comment:
This is a hacky fix, but it works. Thoughts?

Reply:
Hey @barrywhart would you mind pushing this to a branch on the sqlfluff repo? I'd like to clone it and take a look